### ISSUE TITLE
The units for precipitation

### ISSUE THREAD

**Issue by smartlixx** [pacificclimate/climdex.pcic.ncdf#12]
Many thanks for your great work.

When I use this tool for rainfall data from observation, I surprisingly find that it requires a unit of kg m-2 d-1, and it can't take mm. I checked the code but can't find an option to change this requirement. Is it possible to make this controlled by a switch? After all, people may work on model outputs or observational datasets.

**Comment by rod-glover** [comment]
TL;DR: mm/d and kg/m<sup>2</sup>/d are the same unit, given that it is kg of _water_.

Our data is from model outputs, and kg m-2 d-1 is how the units of precipitation are expressed from them. It's a simplified (no fancy formatting) form of the more standard scientific units notation that reads kg m<sup>-2</sup> d<sup>-1</sup> or kg/m<sup>2</sup>/d.

I'm not sure if your problem is that the format of these units is not acceptable to some code you are using, or if you are concerned about the units themselves. However, the following explanation might help in either case:

kg m<sup>-2</sup> d<sup>-1</sup> or kg/m<sup>2</sup>/d is exactly equivalent to mm/d (of water):

1 kg of water has a volume of 1 l.
1 l = 0.001 m<sup>3</sup>
So, for water, 1 kg/m<sup>2</sup> = 0.001 m<sup>3</sup>/m<sup>2</sup> = 0.001 m = 1 mm
and 1 kg m<sup>-2</sup> d<sup>-1</sup> = 1 mm/d

Point being, these two units of measure are synonyms.

**Comment by corviday** [comment]
I believe the issue is that @smartlixx has observation-derived precipitation data with units in mm/day, and would like to calculate Climdex indices with that data using this package, but cannot do so because the package requires the units of the precipitation variable to be kg m-2 d-1.

**Comment by smartlixx** [comment]
> I believe the issue is that @smartlixx has observation-derived precipitation data with units in mm/day, and would like to calculate Climdex indices with that data using this package, but cannot do so because the package requires the units of the precipitation variable to be kg m-2 d-1.
> 
> @smartlixx, are you able to just rename the units in your data files and have it work?

Yes, you are right.

@corviday I just use nco to modify the units and it works now. But it would still be preferable to have a switch for this.

**Comment by corviday** [comment]
Thanks, that's a good idea. Unfortunately, I probably won't have time to work on it for a while, so I'm glad you're able to make things work with nco for now.
